src/main/java/com/gojek/beast/protomapping/ProtoUpdateListener.java

```diff
@@ -65,20 +68,19 @@ public ProtoUpdateListener(ProtoMappingConfig protoMappingConfig, AppConfig appC
 
     private void createStencilClient() {
         if (protoMappingConfig.isAutoSchemaUpdateEnabled()) {
-            stencilClient = StencilClientFactory.getClient(appConfig.getStencilUrl(), System.getenv(), Stats.client().getStatsDClient(), this);
-            this.bqClient.setStencilClient(stencilClient);
-            log.info("updating bq table at startup for proto schema {}", getProto());
-            try {
-                updateProtoParser();
-            } catch (ProtoNotFoundException | BigquerySchemaMappingException e) {
-                String errMsg = "Error while updating bigquery table:" + e.getMessage();
-                log.error(errMsg);
-                e.printStackTrace();
-                throw new BQTableUpdateFailure(errMsg);
-            }
-
+            stencilClient = StencilClientFactory.getClient(stencilConfig.getStencilUrl(), System.getenv(), Stats.client().getStatsDClient(), this);
         } else {
-            stencilClient = StencilClientFactory.getClient(appConfig.getStencilUrl(), System.getenv(), Stats.client().getStatsDClient());
+            stencilClient = StencilClientFactory.getClient(stencilConfig.getStencilUrl(), System.getenv(), Stats.client().getStatsDClient());
+        }
+
+        log.info("updating bq table at startup for proto schema {}", getProto());
+        try {
+            updateProtoParser();
+        } catch (ProtoNotFoundException | BQSchemaMappingException e) {
+            String errMsg = "Error while updating bigquery table:" + e.getMessage();
+            log.error(errMsg);
+            e.printStackTrace();
```

Comment:
clean up needed with these stack traces. Alternatively you could stringify and out log it into the error itself.

Reply:
There are multiple places where we have done so, it gives the whole stack trace